go/types/value_encoder.go

```diff
@@ -129,7 +130,12 @@ func (w *valueEncoder) writeValue(v Value) {
 	case BoolKind:
 		w.writeBool(bool(v.(Bool)))
 	case NumberKind:
-		w.writeNumber(v.(Number))
+		n := v.(Number)
+		f := float64(n)
+		if math.IsNaN(f) || math.IsInf(f, 0) {
```

Comment:
We want to treat `-0` as `0`. But I don't see an actual test for that. Like you said `0 === -0` (in both Go and JS) so the assertRoundTrips test is not testing that.